.github/workflows/docker.yml

```
@@ -10,12 +10,32 @@ on:
 jobs:
   build-and-push:
     runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    env:
+      IAM_ROLE_SESSION_NAME: geodesic-ci
+      AWS_REGION: us-east-1
+      ECR_REGISTRY: public.ecr.aws/ # Images will be published to `public.ecr.aws/cloudposse/$repositoryName`
     strategy:
       matrix:
         os: ['alpine', 'debian']
     steps:
     - name: "Checkout source code at current commit"
       uses: actions/checkout@v2
+
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        role-to-assume: ${{ secrets.ECR_AWS_ROLE }}
+        aws-region: ${{ env.AWS_REGION }}
+        role-session-name: ${{ env.IAM_ROLE_SESSION_NAME }}
+
+    - name: Login to Public Amazon ECR
+      run: |
+        docker logout public.ecr.aws
+        aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws
+
```

You could also add the metadata action

```suggestion

      - name: Docker meta
        id: meta
        uses: docker/metadata-action@v4
        with:
          images: cloudposse/geodesic
```

---

I propose we make this update in a future PR similar to @osterman's comment below

---

sure but the tagging is going to be much more complicated than adding the labels. I would recommend 2 separate PRs. 

1. PR to add labels and metadata action
2. PR to modify the complicated tags logic below to use metadata output